eventuals/grpc/server.h

```diff
@@ -746,8 +747,10 @@ auto UnaryEpilogue(ServerCall<Request, Response>& call) {
               return ::grpc::Status(::grpc::UNKNOWN, e.what());
             })
       | Then([&](auto&& status) {
-           return call.Finish(status)
-               | call.WaitForDone();
+           return call.Finish(status);
+         })
+      | Finally([&](auto&& expected) {
+           return call.WaitForDone();
          });
```

Comment:
Fix up here and below please! By embedding the `Finally()` we are more easily able to catch mistakes where by accident an error would propagate and we haven't properly done a catch, because we only know about the error propagating from `Finish()` so let's bind the catch closely!

```suggestion
           return call.Finish(status)
               | Finally([&](std::optional<std::exception_ptr>&&) {
                    return call.WaitForDone();
                });
```

Note that we really should improve the `Finally()` API to not give us an exception_ptr but instead to give us the equivalent of doing a `Catch().raised<std::exception>())` so that we get a `std::optional<std::exception>` so that we can easily just do an `exception->what()` to log the error message. Can you update `Finally()` in a different review please? Thanks!